op-challenger/game/fault/agent.go

```diff
@@ -67,49 +67,60 @@ func (a *Agent) Act(ctx context.Context) error {
 	if a.tryResolve(ctx) {
 		return nil
 	}
+
 	game, err := a.newGameFromContracts(ctx)
 	if err != nil {
 		return fmt.Errorf("create game from contracts: %w", err)
 	}
 
-	// Calculate the actions to take
 	actions, err := a.solver.CalculateNextActions(ctx, game)
 	if err != nil {
 		a.log.Error("Failed to calculate all required moves", "err", err)
 	}
 
-	// Perform the actions
+	a.syncAct(ctx, actions)
+	return nil
+}
+
+// syncAct parallelizes game actions with a wait group
+func (a *Agent) syncAct(ctx context.Context, actions []types.Action) {
+	var wg sync.WaitGroup
+	wg.Add(len(actions))
 	for _, action := range actions {
-		actionLog := a.log.New("action", action.Type, "is_attack", action.IsAttack, "parent", action.ParentIdx)
-		if action.Type == types.ActionTypeStep {
-			containsOracleData := action.OracleData != nil
-			isLocal := containsOracleData && action.OracleData.IsLocal
-			actionLog = actionLog.New(
-				"prestate", common.Bytes2Hex(action.PreState),
-				"proof", common.Bytes2Hex(action.ProofData),
-				"containsOracleData", containsOracleData,
-				"isLocalPreimage", isLocal,
-			)
-			if action.OracleData != nil {
-				actionLog = actionLog.New("oracleKey", common.Bytes2Hex(action.OracleData.OracleKey))
+		action := action
```

Comment:
It's really just personal preference but I would have had this loop still in the original `Act` method and then split the contents of the `go func()` into a separate method.  You get the `action := action` aliasing automatically because it's passed as a function arg and avoid the extra nesting.

If you prefer to keep this method, I'd probably rename it `performActions`.

Reply:
Given you're on leave, I've just been opinionated and done it my way. :)